include/seqan3/alignment/pairwise_alignment_edit_distance_unbanded.hpp

```
@@ -0,0 +1,551 @@
+            carry_d0 = (carry_d0 != (word_type)0) ? t <= vp : t < vp;
+
+        x = (hp << 1) | (with_overflow_check ? carry_hp : hp0);
+        vn = x & d0;
+        vp = (hn << 1) | ~(x | d0) | (with_overflow_check ? carry_hn : 0);
+
+        if constexpr(with_overflow_check)
+        {
+            carry_hp = hp >> (word_size - 1);
+            carry_hn = hn >> (word_size - 1);
+        }
+    }
+
+    /*!\todo describe me
+     */
+    inline void advance_score(word_type P, word_type N, word_type mask)
+    {
+        if ((P & mask) != (word_type)0)
+            _score++;
+        else if ((N & mask) != (word_type)0)
+            _score--;
+    }
+
+    /*!\todo describe me
+     */
+    inline bool prev_last_active_cell()
+    {
+        score_mask >>= 1;
+        if (score_mask != (word_type)0)
+            return true;
+
+        last_block--;
+        if (is_global && last_block == (unsigned)-1)
+            return false;
+
+        score_mask = (word_type)1 << (word_size - 1);
+        return true;
+    }
+
+    /*!\todo describe me
+     */
+    inline void next_last_active_cell()
+    {
+        score_mask <<= 1;
+        if (score_mask)
+            return;
+
+        score_mask = 1;
+        last_block++;
+    }
+
+    /*!\todo describe me
+     */
+    inline bool update_last_active_cell()
+    {
+        // updating the last active cell
+        while (!(_score <= max_errors))
+        {
+            advance_score(vp[last_block], vn[last_block], score_mask);
+            if(!prev_last_active_cell())
+                break;
+        }
+
+        if ((score_mask == last_score_mask) && (last_block == vp.size() - 1))
+            return on_hit();
+        else
+        {
+            next_last_active_cell();
+            advance_score(vp[last_block], vn[last_block], score_mask);
+        }
+
+        return false;
+    }
+
+    /*!\todo describe me
+     */
+    inline bool on_hit()
+    {
+        // _setFinderEnd(finder);
```

maybe we can throw an exception if some one wants to access this function.

---

we should add a `on_hit` to the configuration which will be called from within this function